### DISCUSSION TITLE
Get helmfile command option sync or destroy

### DISCUSSION THREAD

**Discussion by kiran312-byte** [helmfile/helmfile#854] [Questions]
I would like to execute a script using exec only when executing helmfile sync. I do not want to execute script when executing helmfile destroy. Is there a way I can capture sync or destroy argument in a variable and use conditional if statement

{{ if eq "arg passed" "sync" }}
{{ $synctest := exec "./sync.sh" (list "" ) }}
{{ end }}

**Comment by yxxhero** [comment]
maybe we should introduce a new var {{ .Helmfile.SubCommand }}?

**Comment by xiaomudk** [comment]
You can try to use the hook command:

* `prepare`
* `preapply`
* `presync`
* `preuninstall`
* `postuninstall`
* `postsync`
* `cleanup`

docs: https://helmfile.readthedocs.io/en/latest/#hooks

**Reply by kiran312-byte** [reply]
thanks.I was planning to use hook instead
